### ISSUE TITLE
bench(modelbench): emit live canonical Qwen3.8 Vulkan receipts from raw-decode

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12096]
Parent campaign: #11572. Supports #11959, #11963, and comparison ledger #12093.

Dependencies: #12034 must bind source/archive/binary/device execution and reject empty or missing-parity receipts. Exact model execution also depends on the relevant open capacity/correctness blockers (#11917, #12026, #12082) or an explicit typed refusal showing they are outside the selected artifact/path.

## Classification

- Portfolio tier: 2 (serving); enables tier 1 agentic measurements.
- Centrality: Core
- P1: advanced - create the missing physical, source-bound full-model Qwen3.8 Vulkan receipt for the reachable Strix Halo.
- P2: preserved - subkernel validation and modeled roofline packets remain explicitly unable to establish product-path throughput or quality.
- P3: preserved - connect the shipped real raw-decode runner to the existing canonical Vulkan receipt; do not create another benchmark framework.
- P4: advanced - source, artifact, hardware, execution path, output, performance, and resource evidence fail closed.

## Value frame

- **For:** maintainers deciding which measured Strix Halo bottleneck to optimize.
- **Problem:** `cmd/modelbench/raw_decode.go` executes a real autoregressive path, while its report omits much of the canonical physical identity and counter contract already represented by `internal/compute/qwen38_vulkan_decode_receipt.go`.
- **Today:** the Qwen index has no accepted AMD/Vulkan full-model row; existing Strix headlines are subkernel/synthetic/unbound and receive no product-performance credit.
- **Better because:** one live receipt can select the next kernel lever and can be consumed by the matched scoreboard without hand-transcribing logs.
- **Witness:** a physical raw-decode command on the verified 64GB Ryzen AI Max+ 395 target emits a schema-valid, hash-bound, quality-complete receipt with `engine=fak-native`, executed backend/path, and `fallback_count=0`.

## Exact seams

- Real runner: `cmd/modelbench/raw_decode.go:207-478`.
- Current report gap: `cmd/modelbench/raw_decode.go:541-589`.
- Canonical receipt contract: `internal/compute/qwen38_vulkan_decode_receipt.go:24-238`.
- Quality oracle: `internal/quality/oracle.go`; immediate live numeric oracle may use the raw-decode CPU verifier when the trace runner is not model-bound.

## Required receipt fields

Full source commit, source-archive SHA256, binary SHA256, clean/dirty state, model/artifact and tensor-inventory SHA256, tokenizer/template and exact prompt/output token IDs, quantization, context/output lengths, sampler/seed/EOS, OS/kernel/Mesa/Vulkan/firmware/device, clocks/power/thermal profile, actual RSS/GTT/free-memory/swap, dispatch/transfer/tensor-home/fallback counters, raw per-repetition prefill/decode/total timings, finite logits, CPU/model parity, output text, and capture command.

Missing source binding, engine identity, token counts, quality, counters required for a claimed ratio, or physical execution returns `UNAVAILABLE`/`NOT_COMPARABLE`; it never receives a passing default.

## Execution protocol

- Verify the changed SSH host fingerprint out of band before access; never disable host-key checking or delete trust state.
- Do not restart or reconfigure the active Q4_K_M service without ownership handoff. Use a separately admitted sidecar or maintenance window.
- Three uncounted warmups followed by at least five uncontended measured repetitions; preserve every run and failure.
- Report median, min/max, CV, raw samples, accepted output tok/s, prefill, TTFT/TPOT when actually observed, peak memory, thermals, and no-fallback identity. CV must be <=5% for promotion.

## First checkable step

Add a failing round-trip test proving that a raw-decode result missing source/binary/artifact/device/engine/output identity cannot become a canonical physical receipt; then wire existing result fields without inventing unavailable metrics.

## Witness

```text
go test ./cmd/modelbench ./internal/compute -run "Test.*(RawDecode|Qwen38VulkanDecodeReceipt)" -count=1
```

Then run one bounded B=1, short-output physical smoke on the verified Strix target and attach the raw receipt. The smoke proves plumbing, not a comparison win.

## In scope

Wire observed raw-decode identity, quality, timing, and resource fields into the canonical receipt and capture one bounded physical smoke.

## Dependencies

- After: #12034

## Out of scope

No kernel tuning, service restart, modeled counters, comparison ratio, or performance-win claim.

## Done condition

The fail-closed round-trip tests pass and a verified-host B=1 physical smoke emits a source/binary/artifact/device/engine-bound canonical receipt.

## Work unit

leaf

## Expected steps

6

```routing
lane: modelbench-strix-receipt
paths: cmd/modelbench/raw_decode.go, cmd/modelbench/raw_decode_test.go, internal/compute/qwen38_vulkan_decode_receipt.go, internal/compute/qwen38_vulkan_decode_receipt_test.go
expected_steps: 6
```

**Comment by anthony-chaudhary** [comment]
Device-free canonical raw-decode receipt bridge landed and pushed in cdfb66ee139209229237b42ed9d0cca4fb6fb5f5 (cmd/modelbench r50+gcdfb66ee1; internal/compute r360+gcdfb66ee1).

- V2 promotion fails closed unless source/archive/binary/tree state, model/artifact/inventory/tokenizer/template, exact device/runtime/engine/fallback, token/output, finite-logit, CPU/model parity, per-run controls, timings, memory, and named counters are observed and valid.
- Failure returns a literal zero receipt. Synthetic and merely Vulkan-named paths remain explicit UNAVAILABLE/non-credit and do not gain fak-native identity.
- Candidate timing excludes CPU verification time; the oracle duration remains separately recorded.
- Cross-receipt comparison now binds repetition order and request shape (context/generation/sampler/seed/EOS) while allowing measured performance to differ.

Witnesses: two independent review cycles, final ACCEPT; root focused tests PASS; agent full internal/compute + cmd/modelbench tests PASS; both vets PASS; diff-check PASS; dos commit-audit OK/diff-witnessed; safe reconcile/push complete.

#12096 stays OPEN: modelbench currently lacks the remaining live provenance/telemetry sources needed to promote an AVAILABLE receipt, and the verified-host B=1 physical smoke is still pending. This commit establishes the honest bridge; it is not a throughput result.

**Comment by anthony-chaudhary** [comment]
Physical readiness audit remains NO-GO for a credited B=1 run:

- #12096 live canonical raw-decode promotion is still incomplete.
- Current origin/main physical fanout is non-credit because it attaches a runner whose evidence is not observed device telemetry; #12097 contains the fail-closed repair but it is unlanded.
- #12139 contains the accepted five-pair statistical admission verifier but is also unlanded behind shared-trunk owner reconciliation.
- The appliance has no verified pinned AMD/Vulkan llama.cpp comparator for the frozen 32K, five-pair envelope.
- Host-key attestation and service ownership are still missing under #12034.

Do not run fak bench subagent --simulated=false for evidence from current origin/main. After trust confirmation, a one-token request that preserves the same systemd InvocationID, PID, and health may prove service liveness only. It earns zero Vulkan throughput, quality, or comparison credit.

**Comment by anthony-chaudhary** [comment]
Independent adversarial review rejected the attempted `cmd/modelbench` promotion-helper leaf.

Confirmed: the current production path remains fail-closed because partial runner observations cannot satisfy `BuildQwen38VulkanDecodeReceipt`.

Rejected claim: extracting the existing builder call into a helper did not capture any missing provenance. Its positive test manually filled source, artifact, device, runtime, fallback, output, memory, and dispatch fields, so it proved only that a synthetically complete struct validates. It did not prove those values were observed by the real modelbench path or structurally exclude CLI/environment laundering.

The two-file experiment was removed after review. The next leaf for #12096 must wire one real, runner-owned observation boundary (with a fail-before/pass-after test) rather than adding another synthetic builder fixture.

**Comment by anthony-chaudhary** [comment]
Accepted bounded prerequisite leaf: modelbench now captures the running binary SHA-256 solely from `os.Executable()` plus the existing `fileIdentity` implementation. There is no argv, flag, environment, caller-supplied path/digest, or fallback channel. Capture failure leaves the field empty, and all still-missing provenance keeps the physical receipt `UNAVAILABLE` and non-creditable.

Independent review: ACCEPT. Independent direct-WSL witness:

`go test ./cmd/modelbench -run '^TestRawDecodePhysicalReceiptCapturesRunningBinary$' -count=1` => PASS (`0.031s`).

Exact two-path checkpoint: `refs/fak/wip/codex-goal-strix-raw-receipt`, object `1118d92bf125fdab433d3f62c30147ed9b68dab0`, buildable=true. This is one real observed field, not completion of #12096.

**Comment by anthony-chaudhary** [comment]
The real fanout adapter prerequisite landed in `20e60b4d4`: #12097 now reaches `internal/rawdecode` from explicit model/artifact/backend/token inputs and no longer contains the synthetic `ProductPhysicalRunner` path.

The adapter deliberately leaves source/archive/binary, device, memory, fallback, queue/prefix-reuse, and counter evidence unavailable. Physical validation consequently fails before any receipt bytes are emitted. This confirms the next #12096 leaf must be a runner-owned observer contract, not caller-supplied or environment-relabelled identity. No hardware run occurred.

**Comment by anthony-chaudhary** [comment]
Progress 2026-09-08: the next runner-owned provenance leaf landed on public main in commit 3c4ac13650.

- cmd/modelbench now hashes the bytes at os.Executable() through the existing descriptor-scoped fileIdentity helper and maps only that observed SHA-256 into Source.BinarySHA256.
- No command-line flag, environment value, caller path, or supplied digest can populate or override this field.
- Binary-only evidence remains UNAVAILABLE, non-creditable, with a nil canonical receipt.
- A mapper regression and a production JSON round-trip regression bind the reported digest to an independent hash of the running test executable and confirm that source/archive/tree/device/memory/counter identity remains absent.

Post-reconcile witnesses:
- go test ./cmd/modelbench ./internal/compute -run 'Test.*(RawDecode|Qwen38VulkanDecodeReceipt)' -count=1: PASS
- go vet ./cmd/modelbench ./internal/compute: PASS
- exact commit public leak audit: clean

This is one observed provenance field, not source-to-binary proof and not a physical performance result. On Linux, final credit still requires immutable staged-binary ownership or equivalent executed-image binding rather than assuming the executable pathname cannot be replaced. The issue remains open for source/archive/tree binding, model/device/runtime/fallback/output/memory/counter observation, trusted-host execution, and the physical smoke. No appliance was contacted or modified.

**Comment by anthony-chaudhary** [comment]
[SW-VERIFIED] The first runner-owned provenance prerequisite is now on public `main`: `1e935bc93021941f9de2d6a071a7f8634b8807d6` adds a sealed, zero-input observer for the running binary's full VCS revision, observed clean/dirty state, executable byte count, and executable SHA-256. It rejects missing/malformed build metadata and ambiguous executable reads with a zero result. Focused tests/vet, exact-path validation, and leak audit are green.

This does not attach partial identity to a physical receipt. Source archive, exact model inventory/tokenizer/template, executed device/runtime/fallback, memory, dispatch/transfer, and tensor-home observations are still unavailable, so physical JSON/human output remains fail-closed and zero bytes.

The next bounded software leaf is governed as #12213 (executed Vulkan identity plus monotonic backend-owned counter deltas), with its route landed in `82afb4f3fa246cb4ac9fcd33b5662be755f2d132`. No appliance, service, SSH, or physical hardware was accessed. This update earns no throughput or hardware-witness credit; this issue remains open.

**Comment by anthony-chaudhary** [comment]
[SW-VERIFIED] The backend-owned observation leaf from #12213 is now on public `main` at `8f99aceabe6abc2f43cb61e452bfda0c8963dc1a`.

Raw decode now captures selected-backend identity and per-repetition deltas for observed dispatch, transfer, Q4_K stage/fallback, tensor-home, and device-memory state. Unsupported or incomplete observers remain absent; identity drift, counter reset/wrap, and forged resolver labels fail with a zero result. Focused compute/rawdecode tests and both-package vet are green after the final rebase.

This does not promote a physical receipt. The fanout adapter does not consume the new observation, source-archive and exact model/host identity remain unavailable, native Vulkan build awaits an SDK-equipped host, and no trusted-host smoke was run. No benchmark or service action occurred. This issue remains open and earns no physical performance credit.

Detailed software witness and outstanding gates: #12213 comment 5587786869.

**Comment by anthony-chaudhary** [comment]
#12232 landed on origin/main as 5b09cba2ff13f8386035591097c3049a0a129c2e.

[SW-VERIFIED] modelbench now preserves backend-owned per-run observations and maps only exact executed device, Vulkan runtime, backend, execution path, and fallback fields into the canonical physical attempt. It does not zero-fill peak memory or the missing H2D/D2H counts and D2D bytes required by the canonical counter schema, so Counters remains nil and the receipt remains UNAVAILABLE.

The source tuple is also fully absent pending the reopened #12211 explicit vcs=git parser guard. Full cmd/modelbench and internal/qwen38campaign tests plus vet passed; no hardware/service was accessed. This parent remains open for complete source/model/host/memory identity and the trusted-host smoke.

**Comment by anthony-chaudhary** [comment]
Fresh integration audit: #12195 and the new Vulkan/build observations are still dead to the canonical receipt path. internal/rawdecode.Run stores BackendExecution, but cmd/modelbench rawDecodePhysicalReceipt reads only Backend.Selected; Engine.Name/Runtime/ExecutedPath/FallbackCount, Device, Counters, and device memory remain absent, so BuildQwen38VulkanDecodeReceipt stays UNAVAILABLE. f21eb416d ComputeBuildReceipt has no modelbench/rawdecode/qwen38 consumer and SourceArchiveSHA256 remains empty. Smallest next leaf here: bind rawdecode BackendExecution plus the clean source/archive/SPIR-V/artifact build receipt into rawDecodePhysicalReceipt, then prove field mutations fail closed. No physical credit yet.

**Comment by anthony-chaudhary** [comment]
[SW-VERIFIED] Landed `4fd9a4eaed2ba1219ace7451fb0ae30f4d4af3c4` on `origin/main`.

`cmd/modelbench` now keeps the backend-observed Vulkan API identity exclusively in `device.vulkan_version` and records the canonical fak-native engine runtime as `native`. The focused regression also proves that the otherwise incomplete physical envelope remains `UNAVAILABLE`, non-creditable, and has no receipt.

Verification:
- focused runtime-separation and fail-closed tests: PASS
- `go test ./cmd/modelbench -count=1`: PASS
- `go vet ./cmd/modelbench`: PASS
- exact staged path/diff and public leak audits: PASS
- `fak validate --mine`: infrastructure gate did not complete; WSL preflight timed out, and the native retry timed out during `list_graph` before build/test selection. The package test and vet were run directly and passed.

No fanout phases, hardware, service, appliance, or performance evidence were changed or accessed. #12096 remains open for the rest of the canonical identity, memory, counter, output, and trusted-host witness envelope.

**Comment by anthony-chaudhary** [comment]
SW-VERIFIED producer landed on public `main` in `8854f137b9668f09af3ddd8a5cb4a4d6d5fa061a`.

- `ObserveVulkanHostEnvironment` accepts an actual compute backend, obtains its backend-owned Vulkan identity, and cross-binds its exact device name plus AMD PCI vendor/device IDs to one complete RADV `vulkaninfo --summary` block and one matching DRM sysfs device.
- OS, architecture, and kernel are cross-checked between fixed `/usr/bin/uname -srm`, the running binary target, and procfs. Mesa driver/version and VBIOS firmware come only from command/sysfs observations. No flag, argv field, environment label, package-version fallback, inferred GPU, or static hardware default can populate the tuple.
- Every unsupported, missing, malformed, duplicate, ambiguous, cancelled, or mismatched observation returns an error and the zero tuple. The result is supplementary environment evidence, not execution authority or physical credit by itself.

Device-free witness:

```text
go test ./internal/compute -run '^TestObserveVulkanHostEnvironment' -count=1
ok github.com/anthony-chaudhary/fak/internal/compute
go vet ./internal/compute
PASS before final disjoint trunk fast-forward
python tools/scrub_public_copy.py --audit-staged --root <isolated-public-worktree>
PASS
git diff --cached --name-only
internal/compute/host_environment.go
internal/compute/host_environment_test.go
```

The focused test passed again after rebasing onto the then-current `origin/main` and immediately before push. A broad `go test ./internal/compute -count=1` and the post-rebase vet retry produced no output within their bounded local command windows, so no full-suite/post-rebase-vet claim is made; the issue remains open for that final verification rather than weakening its declared done condition.

No hardware, appliance, SSH, service, model, or benchmark action occurred. This commit grants zero physical performance credit and does not close #12096 or #12082.

**Comment by anthony-chaudhary** [comment]
Landed the exact two-file executable-provenance leaf in 0daffb392a21df7d6fbf6bc224edca2ea384f315. It atomically maps only the independently observed VCS revision, modified flag, and running-binary SHA-256; source-archive/diff identity and binary byte count remain absent, so the receipt stays UNAVAILABLE and zero-credit. The focused test and vet commands are currently blocked before modelbench compiles by open #12292 at internal/metalgemm/zerocopy_stub.go:57 and :62 (undefined Q4KWeight). No hardware, service, environment/argv identity, or performance claim.

**Comment by anthony-chaudhary** [comment]
Schema prerequisite #12294 landed on `main` as `075b803c81c506e8f06eae343c994c9591e5c149`.

The opt-in v3 receipt now derives resource aggregates from complete per-run observations across exactly `reported_runs` measured runs: peak memories are independent maxima, transfer/counter totals are overflow-checked sums, observed-zero directions require explicit completeness, and warmups are excluded. V2 remains readable; mixed v2/v3 comparison is refused.

This is `[SW-VERIFIED]` only. Backend producers, runner wiring, and the trusted-host physical smoke remain open, so #12096 is not closed.

**Comment by anthony-chaudhary** [comment]
#12294 post-land review correction is on `main` at `7f1a50ac78b304e4f640fbed66400ece425f0c22`: every v3 resource run must now be explicitly `kind=measured`; warmup/missing/mixed kinds are rejected and v2 remains unchanged. This restores exact measured-run aggregation semantics. No physical claim.
